elyra/pipeline/pipeline_definition.py

```diff
@@ -432,6 +436,23 @@ def validate(self) -> list:
 
         return validation_issues
 
+    def propagate_global_properties(self):
```

Comment:
Is there any difference between "propagating" vs "if not local, read global value"?

Reply:
The approach using propagation isolates the changes to one location, otherwise, you'd have to know the locations where these kinds of properties are used (and probably all the way down to actual submission).  I think validation would be easier as well and it minimizes the risk of errors.

In a side-discussion, you mentioned a use-case where we might need to know whether a given value was inherited from the globals or not.  If that need were to arise, one could compare the node's value to the global's value and, if the same, assume inherited, otherwise not.  The only case that doesn't solve is when the node redundantly defines a property and it's technically not inherited, but that strikes me as fairly "edgy".